DICOM/src/Microsoft.Health.DeID.SharedLib/DateShiftFunction.cs

```diff
@@ -0,0 +1,77 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Text;
+using EnsureThat;
+using Microsoft.Health.Dicom.DeID.SharedLib.Exceptions;
+using Microsoft.Health.Dicom.DeID.SharedLib.Settings;
+
+namespace Microsoft.Health.Dicom.DeID.SharedLib
+{
+    public class DateShiftFunction
+    {
+        public DateShiftFunction(DateShiftSetting dateShiftSetting = null)
+        {
+            dateShiftSetting ??= new DateShiftSetting();
+            DateShiftKey = dateShiftSetting.DateShiftKey;
+            DateShiftKeyPrefix = dateShiftSetting.DateShiftKeyPrefix;
+            DateShiftRange = dateShiftSetting.DateShiftRange;
+        }
+
+        public string DateShiftKey { get; set; }
+
+        public string DateShiftKeyPrefix { get; set; }
+
+        public uint DateShiftRange { get; set; }
+
+        public string ShiftDate(string inputString, string inputDateFormat = null, string outputDateFormat = null, IFormatProvider provider = null)
```

Comment:
🟡 **[Suggestion]**
`ShiftDate` and `ShiftDateTime` are almost the same, return type, parameters, logic.
The only difference is a format string: `DeIDGlobalSettings.DateFormat` and `DeIDGlobalSettings.DateTimeFormat`
If we make it a parameter, then actually they can merged to one function.